inc/functions/files.php

```diff
@@ -818,15 +818,34 @@ function rocket_clean_domain( $lang = '' ) {
 	$urls = apply_filters( 'rocket_clean_domain_urls', $urls, $lang );
 	$urls = array_filter( $urls );
 
+	$cache_path = rocket_get_constant( 'WP_ROCKET_CACHE_PATH' );
+
+	try {
+		$cache = new RecursiveDirectoryIterator( $cache_path, FilesystemIterator::SKIP_DOTS );
+	} catch ( UnexpectedValueException $e ) {
+		// No logging yet.
+		return;
+	}
+
+	try {
+		$iterator = new RecursiveIteratorIterator( $cache, RecursiveIteratorIterator::CHILD_FIRST );
+	} catch ( Exception $e ) {
+		// No logging yet.
+		return;
+	}
+
+	$dirs_to_preserve = get_rocket_i18n_to_preserve( $lang );
+	/** This filter is documented in inc/front/htaccess.php */
+	$url_no_dots = apply_filters( 'rocket_url_no_dots', false );
+
 	foreach ( $urls as $url ) {
 		$file = get_rocket_parse_url( $url );
 
-		/** This filter is documented in inc/front/htaccess.php */
-		if ( apply_filters( 'rocket_url_no_dots', false ) ) {
+		if ( $url_no_dots ) {
 			$file['host'] = str_replace( '.', '_', $file['host'] );
 		}
 
-		$root = WP_ROCKET_CACHE_PATH . $file['host'] . '*' . $file['path'];
+		$root = $cache_path . $file['host'] . $file['path'];
```

Comment:
We've changed the value of `$root` that's being passed to any callbacks registered to both actions. But we are not acting upon `$root`, i.e. only passing it to both action events. 

Since we're not using it, should we retain the original for those callbacks using it? Then we won't break any 3rd party code.

Reply:
I made this change because in fact the `*` is more of a bother than useful when we are hooking on `before_rocket_clean_domain` in other parts of the code.

As an example in here for our Varnish addon: https://github.com/wp-media/wp-rocket/blob/00cf7a1c5e0b2e7354b0aa76ca86603ee4420eae/inc/Addon/Varnish/Varnish.php#L114

I'm removing the `*` before using the value, else it wouldn't work correctly.

I thought it was a good opportunity to get rid of something that didn't serve us all this time. But I can revert it if we feel it's too much.

Reply:
Hmm, interesting point. What do you think is best?

Reply:
I know that internally, we don't have a use for the `*` anywhere when we use `before/after_rocket_clean_domain`.

I don't know if a 3rd party has/had a use for it. But we do know that it was causing an issue for some, like reported in this issue: https://github.com/wp-media/wp-rocket/issues/1527